cli/azd/pkg/auth/cache.go

```diff
@@ -52,6 +93,34 @@ func (a *msalCacheAdapter) Export(ctx context.Context, cache cache.Marshaler, _
 	return a.cache.Set(cCurrentUserCacheKey, val)
 }
 
+// Normalize keys by removing upper-case keys and replacing them with lower-case keys.
+// In the case where a lower-case key and upper-case key exists, the lower-case key entry
+// takes precedence.
+func normalizeKeys(m map[string]json.RawMessage) {
+	for k, v := range m {
+		if hasUpper(k) {
```

Comment:
Since you are calling `string.ToLower()` anyway (below), do you need to call this `hasUpper()` ??

I think you can call `lower := strings.ToLower()` first and then check if `lower != k` (k has UpperCase when different)

Reply:
calling `strings.ToLower()` on each key would result in additional string allocations. Perf doesn't actually matter in the end, but I feel better with the more pragmatic approach.